02-microarray/dimension-reduction_microarray_01_pca.Rmd

```diff
@@ -0,0 +1,307 @@
+
+```{r}
+# Perform Principal Component Analysis (PCA)
+pca <- prcomp(t(df), # We have to transpose our data frame to meet the `prcomp()` function requirements
+  scale = TRUE # This tells R that we want the variables scaled to have unit variance
+)
+
+# Make the first 10 principal components into a data frame for plotting with `ggplot2`
+pca_df <- data.frame(pca$x[, 1:10]) %>%
+  # Turn samples_ids stored as rownames into column
+  tibble::rownames_to_column("refinebio_accession_code") %>%
+  # Bring only the variables that we want from the metadata into this data frame -- here we are going to join by `refinebio_accession_code` values
+  dplyr::inner_join(dplyr::select(
+    metadata, refinebio_accession_code,
+    histology, subgroup
+  ),
+  by = "refinebio_accession_code"
+  )
+```
+
+Before visualizing and interpreting the results, it can be useful to understand the proportion of variance explained by each principal component.
+For example, if PC1 explained 96% of the variance in your data and very clearly showed a difference between sample batches you would be very concerned about your dataset!
+On the other hand, if a separation of batches was apparent in a different principal component that explained a low proportion of variance and the first few PCs explained most of the variance and appeared to correspond to something like tissue type and treatment, you would be less concerned [@CCDL2020-cluster-validation].
+
+The `summary()` function reports the proportion of variance explained by each principal component.
+By accessing the `importance` element, which contains the proportion of variance explained by each principal component, with `<summary results>$importance`, we can use indexing to only look at the first `n` PCs.
+
+```{r}
+# Save the summary of the PCA results using the `summary()` function
+pca_summary <- summary(pca)
+
+# Now access and store the importance information for the first 10 PCs
+pca_importance <- pca_summary$importance[, 1:10]
+
+# Let's take a look at the importance information
+pca_importance
+```
+
+Now that we've seen the proportion of variance for the first ten PCs, let's plot the PC scores for the first two principal components, the components responsible for the most explained proportion of variance in our dataset. 
+
+Let's also label the data points based on their genotype subgroup since this is central to the subgroup specific based hypothesis in the [original paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3683624/#SD1) [@Northcott2012].
+
+```{r}
+# Make a scatterplot using `ggplot2` functionality
+pca_plot <- ggplot(
+  pca_df,
+  aes(
+    x = PC1,
+    y = PC2,
+    color = subgroup # This will label points with different colors for each `subgroup`
+  )
+) +
+  geom_point() + # This tells R that we want a scatterplot
+  theme_classic() # This tells R to return a classic-looking plot with no gridlines
+
+# Print out plot here
+pca_plot
+```
+
```

Comment:
Can we add a sentence after each plot about what we are seeing? Just a quick blurb about interpretation? aka something like `Looks like Group 4 and SHH groups somewhat cluster with each other but____`